main.cpp

```diff
@@ -19,13 +19,23 @@ using namespace clang::tooling;
 
 class CastCallBack : public MatchFinder::MatchCallback {
 public:
-    CastCallBack(Rewriter& rewriter) {
-        // Your code goes here
+    CastCallBack(Rewriter& rewriter):_rewriter(rewriter) {
     };
 
     void run(const MatchFinder::MatchResult &Result) override {
-        // Your code goes here
+	auto cstmtptr = Result.Nodes.getNodeAs<clang::CStyleCastExpr>("cast");
+	auto l_brace = cstmtptr->getLParenLoc();
+	auto r_brace = cstmtptr->getRParenLoc();
+	auto end_of_var = Lexer::getLocForEndOfToken(cstmtptr->getSubExprAsWritten()->IgnoreImpCasts()->getEndLoc(), 0, *Result.SourceManager, LangOptions());

```

Comment:
Please describe the reason of using the`Lexer::getLocForEndOfToken()` function here.
What would be the difference if `rewriter.InsertText(styleCastExpr->getEndLoc().getLocOffset(1), ")");` was used here?
What would be the difference if  `rewriter.InsertTextAfterToken(styleCastExpr->getEndLoc(), ")");` was used here?

Reply:
getLocForEndOfToken() computes the source location just past the end of the token at this source location.

rewriter.InsertText(styleCastExpr->getEndLoc().getLocOffset(1), ")"); give us position right after (cast) and getEndLoc().getLocOffset(1) shifts this position by one to the right.
If we try to use rewriter.InsertText(styleCastExpr->getEndLoc().getLocOffset(1), ")"); and the variable we declare in main has name longer than one symbol like foo we will get f)oo.

If we try to use rewriter.InsertTextAfterToken(styleCastExpr->getEndLoc(), ")"); we will get the same position and output as it seems to me but with the different way.